### PR TITLE
fix(typescript): add missing types

### PR DESCRIPTION
**Pull request by jaulz**
Some helpful properties were missing in the types.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -69,7 +69,8 @@ export interface ClientMetadata {
   authorization_encrypted_response_alg?: string;
   authorization_encrypted_response_enc?: string;
   authorization_signed_response_alg?: string;
-
+  scopes_supported?: string[];
+
```
Comment (panva) [inline types/index.d.ts:73]: `scopes_supported` is not an [IANA registered client metadata property](https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#client-metadata). Please remove this blob.

`scopes_supported` could potentially go to the `IssuerMetadata` interface, but as the library never reacts to its presence it's not included. I'm keen to only include properties that change the client/issuer behaviour. So it's either all or nothing and i think the inclusion of `[key: string]: unknown;` handles all possible metadata for anyone.

File: types/index.d.ts [hunk 2]
```diff
@@ -346,6 +347,7 @@ export class Client {
   [custom.http_options]: CustomHttpOptionsProvider;
   [custom.clock_tolerance]: number;
   metadata: ClientMetadata;
+  issuer: Issuer<this>;
```
Comment (panva) [inline types/index.d.ts:350]: there's also a [`static` version of it](https://github.com/panva/node-openid-client/blob/v3.15.9/lib/client.js#L1565-L1579), please add that too.

**Review by panva — CHANGES_REQUESTED** [review]

File: types/index.d.ts (hunk 2 above)
Comment (jaulz) [reply types/index.d.ts:349]: Done.

**Comment by jaulz** [comment]
Thanks for the detailed review!

File: types/index.d.ts [hunk 3]
```diff
@@ -346,6 +346,8 @@ export class Client {
...
+  static issuer: Issuer<Client>;
```
Comment (panva) [inline types/index.d.ts:350]:
```suggestion
  static issuer: Issuer<TClient extends Client>;
```

File: types/index.d.ts [hunk 4]
```diff
@@ -346,6 +346,8 @@ export class Client {
   [custom.http_options]: CustomHttpOptionsProvider;
   [custom.clock_tolerance]: number;
   metadata: ClientMetadata;
+  issuer: Issuer<this>;
+  static issuer: Issuer<TClient extends Client>;
```
Comment (panva) [inline types/index.d.ts:350]:
```suggestion
  static issuer: Issuer<Client>;
```

**Comment by panva** [comment]
@jaulz thanks for the contribution!

**Pull request merged**